langkit/templates/main_parse_ada.mako

```diff
@@ -199,13 +200,30 @@ procedure Parse is
          Dump_Lexical_Env (Unit);
       end if;
 
+      if Check then
          declare
+            procedure Check_Parents (Node, Parent : ${root_entity.api_name});
```

Comment:
This might not be sufficient to detect duplicated sub-trees. For example, consider the following langkit grammar:
```python
bin_op = Binop(op, Opt("+"), op)
op = List(foo, allow_empty=True)
```
Given an empty string, the parser would generate a Binop with both operands being the same objects (the empty list) but which parents are (rightly) the same.

Reply:
Btw, why is this procedure declared and implemented here?

Reply:
Good point about the sub-trees ! Means that I should check for uniqueness rather than parents consistency. This is what I originally wanted to do but then was lazy declaring a set data type ...